File: exampleProjects/TypescriptExample/src/de/sots/cellarsandcentaurs/domain/model/HitPoints.ts
export class HitPoints {
    current: number;
    max: number;
    temporary: number;

    constructor(current: number, max: number, temporary: number = 0) {
        this.current = current;
        this.max = max;
        this.temporary = temporary;
    }

    static init(max: number): HitPoints {
        return new HitPoints(max, max, 0);
    }
}

File: exampleProjects/PythonExample/src/de/sots/cellarsandcentaurs/domain/models/hit_points.py
from dataclasses import dataclass

@dataclass(frozen=True)
class HitPoints:
    current: int
    max: int
    temporary: int = 0  # Default value for temporary

    @classmethod
    def init(cls, max_value):
        """Class method to create initial HitPoints with max value."""
        return cls(max_value, max_value, 0)